crates/refunder/src/refund_service.rs

```diff
@@ -143,4 +172,123 @@ impl RefundService {
         };
         Ok(result)
     }
+
+    async fn get_ethflow_data_from_db(&self, uid: &OrderUid) -> Result<EncodedEthflowOrder> {
+        let mut ex = self.db.acquire().await.context("acquire")?;
+        let order = read_db_order(&mut ex, uid)
+            .await
+            .context("read order")?
+            .context("missing order")?;
+        let ethflow_order = read_order(&mut ex, uid)
+            .await
+            .context("read ethflow order")?
+            .context("missing ethflow order")?;
+        Ok(order_to_ethflow_data(order, ethflow_order))
+    }
+
+    async fn send_out_refunding_tx(&mut self, uids: Vec<OrderUid>) -> Result<()> {
+        if uids.is_empty() {
+            return Ok(());
+        }
+
+        // only try to refund MAX_NUMBER_OF_UIDS_PER_REFUND_TX uids, in order to fit into gas limit
+        let uids: Vec<OrderUid> = uids
+            .into_iter()
+            .take(MAX_NUMBER_OF_UIDS_PER_REFUND_TX)
+            .collect();
+
+        tracing::debug!("Trying to refund the following uids: {:?}", uids);
+
+        let futures = uids.iter().map(|uid| {
+            let (uid, self_) = (*uid, &self);
+            async move {
+                self_
+                    .get_ethflow_data_from_db(&uid)
+                    .await
+                    .context(format!("uid {uid:?}"))
+            }
+        });
+        let encoded_ethflow_orders: Vec<EncodedEthflowOrder> = stream::iter(futures)
+            .buffer_unordered(10)
+            .filter_map(|result| async {
+                match result {
+                    Ok(order) => Some(order),
+                    Err(err) => {
+                        tracing::error!(?err, "failed to get data from db");
+                        None
+                    }
+                }
+            })
+            .collect()
+            .await;
+
+        self.submitter.submit(uids, encoded_ethflow_orders).await?;
+        Ok(())
+    }
+}
+
+fn order_to_ethflow_data(
+    order: Order,
+    ethflow_order_placement: EthOrderPlacement,
+) -> EncodedEthflowOrder {
+    (
+        H160(order.buy_token.0),
+        H160(order.receiver.unwrap().0), // ethflow orders have always a
+        // receiver. It's enforced by the contract.
+        big_decimal_to_u256(&order.sell_amount).unwrap(),
+        big_decimal_to_u256(&order.buy_amount).unwrap(),
+        Bytes(order.app_data.0),
+        big_decimal_to_u256(&order.fee_amount).unwrap(),
+        ethflow_order_placement.valid_to as u32,
+        false, // ethflow orders are always fill or kill orders
```

Comment:
Very nice catch! 🎉 . I had in mind that the contract does not allow partially fillable orders. But it does. And it might also make sense for future use cases.

THen I will make sure in another PR that we will not refund these, and in the DB-operation refundable_orders.